src/cdk/table/table-layout-strategy.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {InjectionToken} from '@angular/core';
+import {CdkTable} from '@angular/cdk/table/table';
+
+/**
+ * Describes the return type of a {@link _TableLayoutStrategy}. A `null` value indicates the table
+ * should use its default layout for the display mode (native, flex).
+ */
+export type _TableLayout = DocumentFragment | null;
+
+/** Interface for a service that constructs the DOM structure for a {@link CdkTable}. */
+export interface _TableLayoutStrategy {
```

Comment:
I went back and forth on whether the API should just be `getLayout`, or what you have. I think what you have is fine - it's clear that there are two very separate layout types for the table and they should both be considered separately

Reply:
I also went back and forth on this and eventually chose this API for a few reasons:

1. All implementations of a single `getLayout()` method would contain the same boilerplate code:
```
if (/* native table */) { /* native layout */ } else { /* flex layout */ }
```
2. The explicit methods make the intent clearer and prevent erroneously returning the wrong layout for the table type.
3. Some strategies may want to fall back on the default layout (e.g. scrollable body strategy is only supported for flex layouts, so the default layout is used for native tables).